### DISCUSSION TITLE
共享农场第5期优化升级公告

### DISCUSSION THREAD

**Discussion by otto-001** [famland/website#149] [General]
根据YLE共享农场试运营期情况，为更好地促进共享农场生态发展，社区对共享农场部分参数和运作机制进行了优化和改进。计划于近期对相关合约进行优化调整。升级调整内容如下：

## 5.1 基本田做市手续费优化完善。

取消：基本田二次做市收取做市本金100%手续费规则。

调整为：

对于未做市过famland， 或仍在做市且有基本田做市额度的famland，基本田累加做市资金，无手续费。

对于基本田做市后并撤市的famland，将在首次任何做市资金做市时，一次性支付50USDT+50USDT等额YL的固定手续费，若基本田仍有做市额度，后续累加做市资金则无手续费。

> 例如：
> 1、一个已经做市100U本金的基本田，再增加100U做市本金，无手续费;
> 2、一个已经基本田撤市过的famland，再次做市200U价值的USDT/YL本金，将支付50USDT+50USDT等额YL的手续费，后续再做市100U价值的USDT/YL本金，无手续费。

## 5.2 基本田撤市，做市本金赔付规则优化完善。

调整为：基本田撤市时，用户除了得到做市LP TOken对应的资产数额外，还获得的基本田赔付。

若：做市本金损失大于已领取收益时，有赔付。赔付额度=本金损失 - 收益。

若：做市本金损失小于已领取收益时，无赔付；

若：做市无本金损失时，无赔付；

> 例如：一个满额做市的基本田，做市存入价值500U的LPtoken，因为YL价格下跌导致做市本金价值降低到400U。
>
> 如果用户已经主动或被动提取了累计价值60U的收益。此时，用户撤资基本田后，可以获得400U的LPtoken，还可以获得的赔付=500-400-60=40U。

## 5.3 基本田撤市手续费优化完善。

基本田撤市手续费由1%调整到5%。

## 5.4 新增撤销基本田做市对共享值影响的效能

基本田撤市时，将根据撤市本金数额扣减同等数量的共享值。

> 例如：某个Famland的基本田共享值为560，做市本金的原始价值为100U的USDT/YL ，基本田撤市后，该Famland的共享值将为560-100=460.
>
> 如果玩家继续在这个Famland基本田新增做市200U的USDT/YL的本金，那么Famland共享值仍然可以提升200，达到460+200=660。

## 5.5 新增Famland有效共享值对共享农场流速影响

主要是针对早期bug田的。基本田做市本金小于500U的USDT/YL，共享值又高于做市本金的，按实际做市本金计算有效共享值。只有做市本金大于等于500U的，共享值才按实际共享值额度计算流速。

> 例如：Famland基本田做市本金是100U的USDT/YL , 而其共享值为650，那么农场计算共享值时，不按650计算共享值，而是按照100计算有效共享值。如果基本田做市本金达到500U的USDT/YL，则农场计算总共享值时，才按照650计算有效共享值。

## 5.6 农田进出农场收费规则优化完善。

共享农场未做过广告或已做了广告且有效期已截止时，农田加入农场或农田主动从农场退出时，均要支付30YL。

共享农场做了广告且在广告有效期内，农田主动加入农场或农田主动从农场退出时，均要支付3YL。

以上只涉及手续费率调整，手续费的分流规则不变。

## 5.7  共享农场广告费收费规则优化完善。

取消：农场广告到期后广告费，再次做广告翻倍的规则。

调整为：不论是农场首次做广告，还是广告到期再次做广告，广告费率均按以下标准执行：

|广告期限|广告费|日均费用|
| --- | --- | --- |
|30天|600YL|20YL|
|90天|1700YL|19YL|
|180天|2700YL|15YL|
|360天|4300YL|12YL|

**Comment by cryptosoul-vioiv** [comment]
![98245541ae0ab5d26420189b0a8aa32](https://user-images.githubusercontent.com/81346056/197201262-da54938a-5628-4cd4-9c1d-37e57a9f80c0.jpg)
记录一下合约对于优化五更新的时间
